core/unit_test/tstDeepCopy.hpp

```diff
@@ -118,6 +118,89 @@ void testDeepCopy()
     checkDataMembers( dst_aosoa, fval, dval, ival, dim_1, dim_2, dim_3 );
 }
 
+//---------------------------------------------------------------------------//
+// Perform a mirror test.
+void testMirror()
+{
+    // Data dimensions.
+    const int dim_1 = 3;
+    const int dim_2 = 2;
+    const int dim_3 = 4;
+
+    // Declare data types.
+    using DataTypes =
+        Cabana::MemberTypes<float[dim_1][dim_2][dim_3],
+                            int,
+                            double[dim_1],
+                            double[dim_1][dim_2]
+                            >;
+
+    // Create an AoSoA in the test memory space.
+    int num_data = 423;
+    Cabana::AoSoA<DataTypes,TEST_MEMSPACE> aosoa( num_data );
+
+    // Initialize data.
+    float fval = 3.4;
+    double dval = 1.23;
+    int ival = 1;
+    auto slice_0 = aosoa.template slice<0>();
+    auto slice_1 = aosoa.template slice<1>();
+    auto slice_2 = aosoa.template slice<2>();
+    auto slice_3 = aosoa.template slice<3>();
```

Comment:
I really dislike this syntax.  We should consider implementing this as a free function:
```C++
auto slice_0 = slice<0>(aosoa);
auto slice_1 = slice<1>(aosoa);
```

Reply:
we had something like this before. I'd be fine adding one back in as an alternative to the AoSoA member function